### PR TITLE
chore!: Remove `ButtonDeprecated`, `ButtonDeprecatedBase` and `InputStepper`

### PR DESCRIPTION
**Pull request by aresnik11**
### Overview

Removes `ButtonDeprecated` and `ButtonDeprecatedBase` from being exported, they are still used within `AccordionButtonDeprecated` for now and have been moved within that folder, but will be removed when `Accordion` is fully swapped over to `Disclosure`. Also removes an unused and deprecated `InputStepper` and updates `ModalDeprecated` to not use `ButtonDeprecated`.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [GM-736]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. View the [modal deprecated story](https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app/?path=/docs/molecules-modals-modaldeprecated--modal-deprecated)
2. Click "Open"
3. Confirm you see the new icon x-button and that it functions as expected
4. Confirm you do not see stories for Button Deprecated or Input Stepper
5. Visit https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app/?path=/docs/molecules-pagination--controlled-navigation
6. Confirm you can interact with the controlled pagination and it works as expected
7. Visit https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app/?path=/docs/molecules-tabs--controlled
8. Confirm you can interact with the controlled tabs and it works as expected

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38005)  | [Monolith Env](https://pr-38005-monolith.dev-eks.codecademy.com) |
| Mono       | [Mono Link](https://github.com/codecademy-engineering/mono/pull/6946)  | [Mono LE Env](https://tayra.codecademy.com/workspaces/new?PR_ENV=le-pr-6946)   |

[GM-736]: https://codecademy.atlassian.net/browse/GM-736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.1-alpha.b38f5d.0<br/>@codecademy/gamut-kit@0.6.422-alpha.b38f5d.0<br/>@codecademy/styleguide@66.29.1-alpha.b38f5d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6682fca60e7b30634be6460a--gamut-preview.netlify.app>https://6682fca60e7b30634be6460a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6682fca60e7b30634be6460a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.44734c.0<br/>@codecademy/gamut-kit@0.6.425-alpha.44734c.0<br/>@codecademy/styleguide@66.29.5-alpha.44734c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://668d8aa07494d4d9235432e1--gamut-preview.netlify.app>https://668d8aa07494d4d9235432e1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/668d8aa07494d4d9235432e1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.a55b94.0<br/>@codecademy/gamut-kit@0.6.425-alpha.a55b94.0<br/>@codecademy/styleguide@66.29.5-alpha.a55b94.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://668d933e8aa78bd85bddf7fb--gamut-preview.netlify.app>https://668d933e8aa78bd85bddf7fb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/668d933e8aa78bd85bddf7fb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.2d8675.0<br/>@codecademy/gamut-kit@0.6.425-alpha.2d8675.0<br/>@codecademy/styleguide@66.29.5-alpha.2d8675.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://668d97bbfdcccedfd16c266c--gamut-preview.netlify.app>https://668d97bbfdcccedfd16c266c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/668d97bbfdcccedfd16c266c>Deploy Logs</a></p>

File: packages/styleguide/stories/Molecules/Pagination/Pagination.examples.tsx [hunk 1]
```diff
@@ -23,26 +19,29 @@ export const PaginationControlledExample: React.FC<PaginationProps> = (
...
+      setTotalPages(Number(page));
     },
     [setTotalPages]
   );
 
   return (
     <>
       <FlexBox justifyContent="center" mb={24} p={12}>
-        <InputStepper
+        <Input
```
Comment (dreamwasp) [inline packages/styleguide/stories/Molecules/Pagination/Pagination.examples.tsx:31]: could you add labels to these? (current page / total pages)

**Review by dreamwasp — COMMENTED** [review]
lgtm!!! just have the small label note then :shipit:

File: packages/styleguide/stories/Molecules/Pagination/Pagination.examples.tsx (hunk 1 above)
Comment (aresnik11) [reply packages/styleguide/stories/Molecules/Pagination/Pagination.examples.tsx:31]: you got it

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app>https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/668ee75bf3758252a8a2e803>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
🔥

**Pull request merged**